pennylane_lightning_gpu/src/algorithms/AdjointDiffGPU.hpp

```diff
@@ -22,188 +22,149 @@ template <class CFP_t> static constexpr auto getP11_CU() -> std::vector<CFP_t> {
 }
 
 template <class T = double, class SVType>
-void applyGeneratorRX_GPU(SVType &sv, const std::vector<size_t> &wires,
-                          const bool adj = false) {
+void applyGeneratorPauliRot_GPU(
+    SVType &sv, const std::vector<size_t> &wires, const bool adj = false,
+    const std::vector<std::string> &hyperparams = {}) {
+    PL_ABORT_IF(hyperparams.size() != wires.size(),
+                "Incompatible number of Pauli words and wires");
+    for (size_t i = 0; i < wires.size(); i++) {
+        if (hyperparams[i] == "RX") {
+            sv.applyPauliX({wires[i]}, adj);
+        } else if (hyperparams[i] == "RY") {
+            sv.applyPauliY({wires[i]}, adj);
+        } else if (hyperparams[i] == "RZ") {
+            sv.applyPauliZ({wires[i]}, adj);
+        } else {
+            PL_ABORT("Unsupported Pauli word. Allowed characters are I, RX, RY "
+                     "and RZ");
+        }
+    }
+}
+
+template <class T = double, class SVType>
+void applyGeneratorRX_GPU(
+    SVType &sv, const std::vector<size_t> &wires, const bool adj = false,
+    [[maybe_unused]] const std::vector<std::string> &hyperparams = {}) {
```

Comment:
For my own understanding, why are you adding the input parameter hyperparams if it is never really used?

Reply:
I had to do this following the current implementation of gates & generators in Lightning and Lightning-GPU. But, it won't last long as I am fixing this with some updates in both Lightning and Lightning-GPU to adapt this.